### ISSUE TITLE
Writes to value for persistent node encounters `version conflict` 

### ISSUE THREAD

**Issue by JKOK005** [JKOK005/nus-monorepo#32]
When attempting to write to a hash group in /nodes/<hash_group>, zookeeper throws an error stating `Zookeeper.go:115] zk: version conflict`

Current Zookeeper version locally: 3.4.13

**Comment by JKOK005** [comment]
Noted that 
- We can still register ephemeral nodes under /nodes/<hash_group>
- If the node in /nodes/<hash_group> is deleted, we are able to recreate the node from scratch. Issue only happens when the node exists

**Comment by JKOK005** [comment]
Issue: Zookeeper rejecting writes to node as the version number is constantly set to 0. 

Fix: Change version number to -1
`s.conn.Set(from_path, data, 0)` -> `s.conn.Set(from_path, data, -1)`

**Comment by JKOK005** [comment]
From official ZK docs: 
`Every node in a ZooKeeper tree is refered to as a znode. Znodes maintain a stat structure that includes version numbers for data changes, acl changes. The stat structure also has timestamps. The version number, together with the timestamp allow ZooKeeper to validate the cache and to coordinate updates. Each time a znode's data changes, the version number increases. For instance, whenever a client retrieves data, it also receives the version of the data. And when a client performs an update or a delete, it must supply the version of the data of the znode it is changing. If the version it supplies doesn't match the actual version of the data, the update will fail. (This behavior can be overridden. For more information see... )[tbd...]`

TL;DR
- Every update to a node increments the version of the data
- On read, the data's version number is supplied
- On write, we MUST supply the same version number. If not, ZK will throw a `version conflict` error
- This behaviour can be overwritten if we do a write with version no = -1 (from initial tests)
